cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -79,13 +87,53 @@
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  private static NamespaceId configuredNamespace = configureTestNamespace();
+  // boolean is true if we created the namespace; indicates that we are responsible for cleaning it
+  private static Map<NamespaceId, Boolean> registeredNamespaces = new HashMap<>();
+
   private AccessToken accessToken;
 
   @Before
   public void setUp() throws Exception {
     checkSystemServices();
     assertUnrecoverableResetEnabled();
-    assertIsClear();
+
+    boolean createdByUs = false;
+    if (!getNamespaceClient().exists(configuredNamespace.toId())) {
+      getNamespaceClient().create(new NamespaceMeta.Builder().setName(configuredNamespace.toId()).build());
+      createdByUs = true;
+    }
+    registeredNamespaces.put(configuredNamespace, createdByUs);
+    assertIsClear(configuredNamespace);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    for (Map.Entry<NamespaceId, Boolean> namespaceEntry : registeredNamespaces.entrySet()) {
+      Boolean createdByUs = namespaceEntry.getValue();
+      if (createdByUs) {
+        getNamespaceClient().delete(namespaceEntry.getKey().toId());
+      } else {
+        // if we didn't create the namespace, don't delete it; only clear the data/programs within it
+        doClear(namespaceEntry.getKey());
+        assertIsClear(namespaceEntry.getKey());
```

Comment:
can ``assertIsClear`` be called inside ``doClear``, so we don't have to call them separately if ``doClear`` is used elsewhere?